streambed-logged/src/lib.rs

```diff
@@ -32,6 +32,81 @@ const CONSUMER_QUEUE_SIZE: usize = 10;
 const PRODUCER_QUEUE_SIZE: usize = 10;
 const TOPIC_FILE_CONSUMER_POLL: Duration = Duration::from_secs(1);
 
+/// A compaction key is paired with an offset in a map. This map of
+/// compaction keys and their offsets express the minimum offset to
+/// retain as a result of having performed a scan of consumer records.
+#[derive(Clone, Hash, PartialEq, Eq)]
+pub struct CompactionKey {
+    pub topic: Topic,
+    pub partition: u32,
+    pub key: u64,
```

Comment:
I always feel a bit uneasy about seeing basic types like u64 in open code.  In this case it is such an important type I think it should be newtyped, as in struct RecordKey(u64);